core/src/main/java/io/bisq/core/dao/node/consensus/GenesisTxVerification.java

```diff
@@ -48,11 +49,17 @@ public boolean isGenesisTx(Tx tx, int blockHeight) {
     }
 
     public void applyStateChange(Tx tx) {
-        tx.getOutputs().forEach(txOutput -> {
-            txOutput.setUnspent(true);
-            txOutput.setVerified(true);
-            bsqBlockChain.addUnspentTxOutput(txOutput);
-        });
+        long totalOutput = 0;
+        for (int i = 0; i < tx.getOutputs().size(); ++i) {
+            TxOutput txOutput = tx.getOutputs().get(i);
+            totalOutput += txOutput.getValue();
+            // Handle change from genesis transaction, only count outputs below the issuance amount as BSQ outputs
+            if (totalOutput <= bsqBlockChain.getIssuedAmount().getValue()) {
```

Comment:
I would suggest to use the new paradign to do any logic/verification in dedicated classes. Might be a bit of overhead here but will help to isolate the rules from the specs from other code. So to create  GenesisTxOuputsController class where the check is done and if the check return true the state gets updated.